plugins/pulp_docker/plugins/importers/importer.py

```diff
@@ -180,106 +184,148 @@ def import_units(self, source_repo, dest_repo, import_conduit, config, units=Non
             if units_to_add:
-                unit_filter = {'image_id': {'$in': list(units_to_add)}}
-                criteria = UnitAssociationCriteria(type_ids=[constants.IMAGE_TYPE_ID],
-                                                   unit_filters=unit_filter)
-                units = conduit.get_source_units(criteria=criteria)
+                units = models.Image.objects.filter(image_id__in=list(units_to_add))
             else:
                 # Break out of the loop since there were no units to add to the list
                 break
 
-        return units_added
+        return list(units_added)
 
     @staticmethod
-    def _import_manifests(conduit, units):
+    def _import_tag(conduit, unit, dest_repo):
         """
-        Import manifests and referenced blobs.
+        Import a Tag, and the Manifests and Blobs it references.
 
-        :param conduit: provides access to relevant Pulp functionality
-        :type conduit: pulp.plugins.conduits.unit_import.ImportUnitConduit
-        :param units: optional list of pre-filtered units to import
-        :type  units: list of pulp.plugins.model.Unit
-        :return: list of units that were copied to the destination repository
-        :rtype:  list
+        :param conduit:   provides access to relevant Pulp functionality
+        :type  conduit:   pulp.plugins.conduits.unit_import.ImportUnitConduit
+        :param unit:      The Tag to be imported to the repository
+        :type  unit:      pulp_docker.plugins.models.Tag
+        :param dest_repo: The destination repository that the Tag is being imported to. This is
+                          needed because technically we are creating a copy of the Tag there rather
+                          than an association of the Tag, and the repo_id is a required field on the
+                          Tag object.
+        :type  dest_repo: pulp.plugins.model.Repository
+        :return:          list of Units that were copied to the destination repository
+        :rtype:           list
         """
-        units_added = []
+        units_added = set()
 
-        # All manifests if not specified
+        # We need to create a copy of the Tag with the destination repository's id, but other fields
+        # copied from the source Tag.
+        manifest_digests_to_import = set()
+        tag = models.Tag.objects.tag_manifest(repo_id=dest_repo.repo_id, tag_name=unit.name,
+                                              manifest_digest=unit.manifest_digest)
+        units_added.add(tag)
+        conduit.associate_unit(tag)
+        manifest_digests_to_import.add(unit.manifest_digest)
 
-        if units is None:
-            criteria = UnitAssociationCriteria(type_ids=[constants.MANIFEST_TYPE_ID])
-            units = conduit.get_source_units(criteria=criteria)
+        # Add referenced manifests
+        for manifest in models.Manifest.objects.filter(
+                digest__in=sorted(manifest_digests_to_import)):
```

Comment:
In theory it could, but I am not sure it does. To be honest, I just left it that way because the previous code also did this sort. Frankly, if Mongo would benefit from such a sort I'd expect them to do it on their end so this probably isn't necessary.

For now I'd like to leave it because I don't want to disturb the working code so close to the beta.